docs/tutorials/services/constrain-motion.md

```diff
@@ -0,0 +1,408 @@
+Be careful when instructing robot arms to move.
+Before running any code, ensure your robotic arm has enough space and that there are no obstacles.
+Also pay attention to your surroundings, double-check your code for correctness, and make sure anyone nearby is aware and alert before issuing commands to your robot.
+{{< /alert >}}
+
+{{<gif webm_src="../../videos/motion_constraints.webm" mp4_src="../../videos/motion_constraints.mp4" poster="../../videos/motion_constraints.jpg" alt="An arm moving a cup from one side of a tissue box to the other, across a table. The cup stays upright." class="alignright" max-width="250px">}}
+
+Say you want your robot to pass you a cup of tea, but you don't want it to spill the water or bump into other objects on the table.
+
+If you followed along in [part 2 of the Motion Service tutorial series](../plan-motion-with-arm-gripper/), you used the [Motion Service](/services/motion/) to move a robot arm and end effector to desired positions.
+This tutorial builds on that and shows you how to use [constraints](/services/motion/constraints/) and transforms to control the way your robot moves between its start and end position.
+
+In this tutorial, you will learn to move a cup across a table without hitting another object, and while remaining upright.
+
+The [full tutorial code](#full-tutorial-code) is available at the end of this page.
+
+## Prerequisites
+
+{{< alert title="Note" color="note" >}}
+Code examples in this tutorial use a [UFACTORY xArm 6](https://www.ufactory.cc/product-page/ufactory-xarm-6/), but you can use any [arm model](/components/arm/).
+{{< /alert >}}
+
+Before starting this tutorial, make sure you have the [Viam Python SDK](https://python.viam.dev/) <!-- or the [Viam Go SDK](https://pkg.go.dev/go.viam.com/rdk/robot/client#section-readme/) -->installed.
+
+If you are connecting to a real robotic arm during this tutorial, make sure your computer can communicate with the arm controller before continuing.
+
+Familiarize yourself with the concepts outlined in the second motion tutorial, [Plan Motion with an Arm and a Gripper](../plan-motion-with-arm-gripper/), before continuing.
+This tutorial picks up where **Plan Motion with an Arm and a Gripper** stops, so further examples depend on having a connected robot, client and service access, and other infrastructure in place.
+
+For reference, look at [the full code sample from the prior tutorial](../plan-motion-with-arm-gripper/#full-tutorial-code).
+
+## Configure your robot
+
+Use the robot configuration from [the previous tutorial](../plan-motion-with-arm-gripper/) for this tutorial, including the [arm](../../../components/arm/) and [gripper](../../../components/gripper/) components with [frames](../../../services/frame-system/) configured.
+
+The Motion Service is one of the "built-in" services, so you don't need to do anything to enable it on your robot.
+
+{{% expand "Click for an example raw JSON config." %}}
+
```

Comment:
```suggestion

If you go to your robot's **Config** tab on [the Viam app](https://app.viam.com/) and click **Raw JSON**, your config should resemble the following:

```

Reply:
@andf-viam I did add this based on your first round of feedback--not enough?

Reply:
Sure, this works for me, thank you!!